packages/number/src/NumberEditor.tsx

```diff
@@ -20,73 +25,144 @@ export interface NumberEditorProps {
   field: FieldAPI;
 }
 
-type RangeValidation = { min?: number; max?: number };
-
-function getRangeFromField(field: FieldAPI): RangeValidation {
-  const validations = field.validations || [];
-  const result = validations.find((validation) => (validation as any).range) as
-    | { range: RangeValidation }
-    | undefined;
-  return result ? result.range : {};
-}
-
-function valueToString(value: InnerNumberEditorProps['value']) {
-  return value === undefined ? '' : String(value);
-}
-
 type InnerNumberEditorProps = Pick<
   FieldConnectorChildProps<number>,
   'disabled' | 'errors' | 'setValue' | 'value'
 > & {
   field: NumberEditorProps['field'];
 };
 
+const NUMBER_STEP = 1;
+
 function InnerNumberEditor({
   disabled,
   errors,
   field,
   setValue,
   value: sdkValue,
 }: InnerNumberEditorProps) {
-  const previousValue = React.useRef(valueToString(sdkValue));
   const [inputValue, setInputValue] = React.useState(valueToString(sdkValue));
   const range = getRangeFromField(field);
+  const inputRef = React.useRef<HTMLInputElement>(null);
 
   React.useEffect(() => {
-    previousValue.current = valueToString(sdkValue);
+    const stringSdkValue = valueToString(sdkValue);
+    // Update the input value if the SDK value (numeric) changes
+    if (stringSdkValue !== inputValue) {
+      setInputValue(stringSdkValue);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- we want to trigger it only when sdkValue had been changed
   }, [sdkValue]);
 
-  React.useEffect(() => {
-    const stringifiedSdkValue = valueToString(sdkValue);
-    // Update the input value (string) if the SDK value (numeric) changes
-    if (stringifiedSdkValue !== previousValue.current && stringifiedSdkValue !== inputValue) {
-      setInputValue(stringifiedSdkValue);
+  const updateExternalValue = (value: number | undefined) => {
+    if (sdkValue !== value) {
+      setValue(value);
+    }
+  };
+
+  const changeValueByStep = (type: 'increment' | 'decrement') => {
+    const currentValue = isNaN(+inputValue) ? 0 : +inputValue;
+    let nextValue = type === 'increment' ? currentValue + NUMBER_STEP : currentValue - NUMBER_STEP;
+    // Floating point numbers cannot represent all decimals precisely in binary.
+    // This can lead to unexpected results, such as 0.1 + 0.2 = 0.30000000000000004.
+    // See more details: https://floating-point-gui.de/
+    nextValue = +nextValue.toFixed(countDecimals(currentValue));
+
+    setInputValue(valueToString(nextValue));
+    setValue(nextValue);
+  };
+
+  // Keeps focus on the input
+  const handleControlPointerDown: React.PointerEventHandler<HTMLButtonElement> = (event) => {
+    event.preventDefault();
+    inputRef.current?.focus();
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent<any>) => {
+    const keyToFnMap: {
+      [key: string]: () => void;
+    } = {
+      ArrowUp: () => changeValueByStep('increment'),
```

Comment:
this would then become:
```suggestion
      ArrowUp: () => changeValueByStep(StepChangeType.Increment),
```
if we introduce the enum, similarly for the item below